client/gutenberg/editor/controller.js

```diff
@@ -90,13 +86,16 @@ export const loadTranslations = ( context, next ) => {
 				setLocaleData( localeData, domain );
 			}
 		} );
-
-		next();
 	} );
 };
 
-function waitForSelectedSiteId( context ) {
-	return new Promise( resolve => {
+const waitForSiteId = store =>
```

Comment:
The `waitForSiteId` function can be completely removed, as the `siteSelection` page.js handler guarantees that selected site ID has the right value before calling `next()`. It wasn't always so, was fixed recently in #28449.

Reply:
awesome! thanks - this helps us with another related issue as well